lib/Slackbot_worker.js

```diff
@@ -1,17 +1,21 @@
 var Ws = require('ws');
+var _ = require('lodash');
 var request = require('request');
 var slackWebApi = require(__dirname + '/Slack_web_api.js');
 var HttpsProxyAgent = require('https-proxy-agent');
 
 
 module.exports = function(botkit, config) {
+    config = config || {};
     var bot = {
         botkit: botkit,
-        config: config || {},
+        config: _.clone(config),
         utterances: botkit.utterances,
-        api: slackWebApi(botkit, config || {})
+        api: slackWebApi(botkit, _.clone(config))
```

Comment:
You want to use _.cloneDeep. _.clone is a shallow clone so any objects or arrays in the `config` object will be referenced both by the original object and the cloned one. This could lead to the same modification problem you're seeing now.